src/test/java/io/confluent/connect/elasticsearch/integration/SecurityIT.java

```diff
@@ -0,0 +1,161 @@
+package io.confluent.connect.elasticsearch.integration;
+
+import com.google.gson.JsonObject;
+import io.confluent.common.utils.IntegrationTest;
+import io.confluent.connect.elasticsearch.ElasticsearchSinkConnector;
+import io.confluent.connect.elasticsearch.ElasticsearchSinkConnectorConfig;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestClientFactory;
+import io.searchbox.client.JestResult;
+import io.searchbox.client.config.HttpClientConfig;
+import io.searchbox.core.Get;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.nio.conn.ssl.SSLIOSessionStrategy;
+import org.apache.kafka.common.network.Mode;
+import org.apache.kafka.common.security.ssl.SslFactory;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.runtime.SinkConnectorConfig;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorStateInfo;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.apache.kafka.connect.util.clusters.EmbeddedConnectCluster;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+
+import static java.lang.Thread.sleep;
+import static junit.framework.TestCase.assertTrue;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.TASKS_MAX_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.VALUE_CONVERTER_CLASS_CONFIG;
+
+
+@Category(IntegrationTest.class)
+public class SecurityIT {
+  private static final Logger log = LoggerFactory.getLogger(SecurityIT.class);
+
+  private EmbeddedConnectCluster connect;
+
+  private static final String MESSAGE_KEY = "message-key";
+  private static final String MESSAGE_VAL = "{ \"schema\": { \"type\": \"map\", \"keys\": "
+      + "{ \"type\" : \"string\" }, \"values\": { \"type\" : \"int32\" } }, "
+      + "\"payload\": { \"key1\": 12, \"key2\": 15} }";
+  private static final String CONNECTOR_NAME = "elastic-sink";
+  private static final int AWAIT_TASK_STARTUP_MS = 5000;
+  private static final String KAFKA_TOPIC = "test-elasticsearch-sink";
+  private static final String TYPE_NAME = "kafka-connect";
+  private static final int TASKS_MAX = 1;
+
+  @Before
+  public void setup() throws IOException {
+    connect = new EmbeddedConnectCluster.Builder().name("elastic-sink-cluster").build();
+    connect.start();
+  }
+
+  @After
+  public void close() {
+    connect.stop();
+  }
+
+  private void waitForTasks(int minTasks) throws InterruptedException, TimeoutException {
+    long startTime = System.currentTimeMillis();
+    for (; ; ) {
```

Comment:
use TestUtils.waitForCondition instead of this for loop with sleeps. it'll make the code more readable.